eclair-node/src/main/scala/fr/acinq/eclair/api/directives/ExtraDirectives.scala

```diff
@@ -55,6 +52,13 @@ trait ExtraDirectives extends Directives {
   val countFormParam: NameReceptacle[Int] = "count".as[Int]
   val skipFormParam: NameReceptacle[Int] = "skip".as[Int]
 
+  // @formatter:off
+  // We limit default values to avoid accidentally reading too much data from the DB.
+  // We cannot use a val, otherwise the default values would not be recomputed for every request.
+  def fromFormParam(): NameDefaultUnmarshallerReceptacle[TimestampSecond] = "from".as[TimestampSecond](timestampSecondUnmarshaller).?(TimestampSecond.now() - 1.day)
+  def toFormParam(): NameDefaultUnmarshallerReceptacle[TimestampSecond] = "to".as[TimestampSecond](timestampSecondUnmarshaller).?(TimestampSecond.now())
```

Comment:
Nit: did you put trailing parentheses because those methods depend on `TimestampSecond.now()` which has parenthese, therefore . FWIW in theory neither should have parentheses because they are side-effect-free.

Reply:
My rule of thumb is to put parentheses whenever a function isn't a pure function. This isn't a pure function because invoking it repeatedly yields different results. Whenever there's no parenthesis, from the caller's point of view it looks like a `val` that should return the same result every time, don't you think?

Reply:
Makes sense, it's a stricter definition of side effect, probably more correct than mine.